command/license.go

```diff
@@ -0,0 +1,101 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/mitchellh/cli"
+)
+
+var _ cli.Command = &LicenseCommand{}
+
+type LicenseCommand struct {
+	Meta
+}
+
+func (l *LicenseCommand) Help() string {
+	helpText := `
+Usage: nomad license <subcommand> [options] [args]
+	
+This command has subcommands for managing the Nomad Enterprise license.
+For more detailed examples see:
+https://www.nomadproject.io/docs/commands/license/
+
+Install a new license from a file:
+	$ nomad license put @nomad.license
+
+Install a new license from stdin:
+	$ nomad license put -
+
+Install a new license from a string:
+	$ nomad license put "<license blob>"
+
+Retrieve the current license:
+
+	$ nomad license get
+
+Reset the current license:
+	$ nomad license reset
+	`
+	return strings.TrimSpace(helpText)
+}
+
+func (l *LicenseCommand) Synopsis() string {
+	return "Interact with Nomad Enterprise License"
+}
+
+func (l *LicenseCommand) Name() string { return "license" }
+
+func (l *LicenseCommand) Run(args []string) int {
+	return cli.RunResultHelp
+}
+
+func OutputLicenseReply(ui cli.Ui, resp *api.LicenseReply) int {
+	if resp.Valid {
+		ui.Output("License is valid")
+		outputLicenseInfo(ui, resp.License, false)
+		return 0
+	} else if resp.License != nil {
+		now := time.Now()
+		if resp.License.ExpirationTime.Before(now) {
```

Comment:
Also, wonder if expiration should be returned by the API as it's the canonical value.  Avoids dealing with time syncing issues and cli host having a significant time sync.  Not a big concern though.

Reply:
I think the licenses will be valid for long enough that time sync issues won't impact much, this will likely be checked infrequently and pretty far out from actual expiration so I wasn't really concerned with extremely precise timing